core/src/main/java/hudson/model/User.java

```diff
@@ -425,6 +425,18 @@ private LegitimateButUnknownUserDetails(String username) throws IllegalArgumentE
      */
     @Restricted(NoExternalUse.class)
     public @NonNull Authentication impersonate(@NonNull UserDetails userDetails) {
+        if (!userDetails.isAccountNonLocked()) {
```

Comment:
It's not a big an oversight as you'd expect because every single implementation of `UserDetails` returns `true` for all four of these methods. I'm working on supporting LDAP and AD specific attributes that would fill in real data in a separate PR for those plugins.